Native API Gateway Docs (#16365)
hashicorp/consul: website/content/docs/connect/gateways/api-gateway/configuration/tcp-route.mdx

`"tcp-route"`.

#### Values

- Default: none
- This field is required.
- Data type: string value that must be set to`tcp-route`.

### `Name`

Specifies a name for the configuration entry. The name is metadata that you can
use to reference the configuration entry when performing Consul operations,
such as applying a configuration entry to a specific cluster.

#### Values

- Default: Defaults to the name of the node after writing the entry to the Consul server.
- This field is required.
- Data type: string

### `Parents`

Specifies the list of gateways that the route is bound to.

#### Values

- Default: none
- This field is not required.
- Data type: List of map. Each member of the list contains the following fields:
  - `Kind`
  - `Name`
  - `SectionName`

### `Parents.Kind`

Specifies the type of resource to bind to. This field is required and must be
set to `"api-gateway"`

#### Values

- Default: none
- This field is required.
- Data type: string

### `Parents.Name`

Specifies the name of the api-gateway to bind to.

#### Values

- Default: none
- This field is required.
- Data type: string

### `Parents.SectionName`

> Specifies the name of the listener on the api-gateway. If left empty, this will
> bind this route to *all listeners* on the parent gateway

Reviewer: ```suggestion
Specifies the name of the listener defined in the [`api-gateway` configuration](/consul/docs/connect/gateways/api-gateway/configuration/api-gateway) that the route binds to. If the field is configured to an empty string, the route binds to all listeners on the parent gateway.
```

#### Values

- Default: ""
- This field is not required.
- Data type: string

### `Services`

Specifies a TCP-based service the API gateway should route incoming requests
to. You can only specify one service.

#### Values

- Default: none
- This field is not required.

> - Data type: List of map. Each member of the list contains the following fields:

Reviewer: ```suggestion
- The data type is a list of maps. Each member of the list contains the following fields:
```

  - `Name`

### `Services.Name`

> Specifies the list of TCP-based services this should route to. Currently, this must specify at maximum one service.

Reviewer: ```suggestion
Specifies the list of TCP-based services this should route to. You can specify a maximum of one service.
```
